### PR TITLE
Launchpad: Change hover style for "Choose a Plan task"

### PR DESCRIPTION
**Pull request by sharpirate**
#### Summary
There's been a [recent PR](https://github.com/Automattic/wp-calypso/pull/67763) for enabling going back to the **Choose a Plan** task when it has been completed. Since the task has both a badge and a chevron icon there has been some discussion regarding styling. This is how it looks currently:

https://user-images.githubusercontent.com/20927667/190245586-2749354c-b06b-4fd8-88fe-3c2bcbd9e651.mov

#### Proposed changes
This PR improves the styling by removing the chevron icon and changing the badge's color to on `#91CAF2` hover:

https://user-images.githubusercontent.com/20927667/190245690-e5a8b7af-19b6-4a9d-b3be-a7afc1357c80.mov

#### Testing Instructions
1. Create a new site using the Newsletter flow
2. Choose a paid plan
3. Once you get to the Launchpad screen, you should see the new style for the **Choose a Plan** task. It should have no chevron icon and the badge background color should be `#91CAF2` on hover

#### Pre-merge Checklist

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?

Related to [#67667](https://github.com/Automattic/wp-calypso/issues/67667)

### CODE REVIEW

File: client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/task-helper.ts [hunk 1]
```diff
@@ -38,6 +38,7 @@ export function getEnhancedTasks(
 					taskData = {
 						title: translate( 'Choose a Plan' ),
 						keepActive: true,
+						hideChevron: true,
```
Comment (edanzer) [inline client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/task-helper.ts]: I'm wondering if we can derive hideChevron from existing attributes? If I'm reading it right, seems like maybe we hide the chevron if the task is complete and if there is no badge ( `const hideChevron = isComplete && ! badgeText `). If so, we might be able to skip adding the extra task property and just derive the value dynamically in the ChecklistItem component.

**Comment by edanzer** [comment]
Just tested this and looking good. Chevron is gone for the Plans task, and badge color changes as expected on hover. I left a separate question about whether we need a hideChevron property, or if that value can be derived from existing properties.

**Comment by sharpirate** [comment]
Thanks, @edanzer, I've updated the PR. We just need to make sure we don't display the chevron if the task has a badge, so the `hideChevron` was unnecessary

**Review by edanzer — APPROVED** [review]

File: client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/checklist-item.tsx [hunk 2]
```diff
@@ -40,7 +40,8 @@ const ChecklistItem = ( { task }: { task: Task } ) => {
 				{ task.displayBadge && task.badgeText ? (
 					<Badge type="info-blue">{ task.badgeText }</Badge>
 				) : null }
-				{ ! taskDisabled && (
+				{ /* don't display chevron and badge at the same time */ }
+				{ ! taskDisabled && ! task.displayBadge && ! task.badgeText && (
```
Comment (agrullon95) [inline client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/checklist-item.tsx:44]: Is checking `! taskDisabled && ! task.displayBadge` sufficient here?
Do we need to include `task.badgeText` in this conditional?

Comment (sharpirate) [reply client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/checklist-item.tsx:44]: @agrullon95 That's a good question, I included the `task.badgeText` check for the sake of consistency with `line 40`. I am assuming there's no way `task.displayBadge` will be true but `task.badgeText` will be empty. If that is the case I can remove the `task.badgeText` check

**Review by agrullon95 — APPROVED** [review]
Tested and worked with a paid plan.

**Pull request merged**